### DISCUSSION TITLE
Project Update - June 2022

### DISCUSSION THREAD

**Discussion by DeeEmm** [DIY-Flow-Bench/DIY-Flow-Bench#103] [Announcements]
It's been a long time since I looked at this project. Unfortunately I've long been recovering from a spinal injury, which when mixed in with the Covid epidemic and work commitments has meant that the project has taken a bit of a back seat whilst I dealt with day to day life and work.

A recent message piqued my interest in the project again and so I ordered myself a vortex vac motor and decided to get the shield testing completed so that we can get some shields out into the wild and into the hands of those who want them.

![PXL_20220607_043736757](https://user-images.githubusercontent.com/3038710/172340610-a93ce772-5923-4fa6-bd0e-4830ff29519f.jpeg)

Covid has also had an impact on parts availability and there is still a bit of a semiconductor shortage. Well that's not technically true, you can still get most components but in many cases prices have increased by well over 1000%. The impact on this project is largely the pressure sensors as they are also shared with respirators, however now that Covid is settling down the price is starting to level out for many of those devices which I assume is a direct result of over-manufacture. Hopefully this is good news for us and we will once again have cheap pressure sensors.

With my interest in the project reinvigorated I sat down tonight to try and figure out where I got up to with testing and to re-familiarise myself with the code. Having not looked a this for some time it's going to take me a little while to get into the right headspace and get back up to speed. Fortunately everything is pretty well commented. 

One of the main things that was previously holding me back from testing was the lack of a proper test bench, so this is one of the first things I will address. I'm still waiting for a VFD to drive the fan I bought but hopefully that will be here soon. In the mean time I will get my prototype up and running again and build a basic bench to test with.

Another major point on the agenda is to review the current project scope and see what needs to be done, versus what can be pushed back. I seem to recall a bit of project creep, which was fine when I was putting a lot of hours into this and wanted to deliver everything, but I figure that If I can get the basic shield and code in a state that allows the shield to be produced, then the nice-to-have features can wait a little longer. I think it is much better to have a functional project with reduced feature-set than a project that has stalled due to feature creep. To this end I will update the roadmap to reflect the change in direction.

So. This is where things currently stand with the project. The project is not dead, it's just been hard for me to get back to it. Hopefully now I've got a renewed interest in it again I will make sure I set some time aside and take it through to completion. I'm not going to promise any dates at this stage simply as I am currently taking life day-by-day but I will try and post some regular updates on progress here in the discussions section.

/DM

**Comment by DeeEmm** [comment]
Further update...

I've refactored the V2.1 project roadmap and moved all non-essential aspects into a new project named V3 Roadmap. The only remaining aspects left in V2.1 are those that are required for the sensors to be read and displayed on the GUI 

What I have decided to do is get the shield functional as a basic realtime display, which will allow it to work much as a regular bench does and then push out everything else to future releases. This means that the code as it currently stands is not that far away. I did most of the refactoring already and so it mostly just requires testing on the prototype shields. I need to make up some new PCBs to address a I2C voltage issue so final testing will not be completed until I make those boards. The new PCB design was pretty much finished before I took leave of the project so I just need to finish reviewing the design changes and undertake design validation before I can get a new batch of PCBs made. I want to be careful here as it's been such a long time since I did the design I want to make sure I do not overlook something.

Moving the other features out to V3 allows for interim bugfix updates to be made on the V2.X project line without having to refactor everything again. Once we have a fully tested shield and basic code I will then move on to the V3 stuff.  My plan there is have smaller more frequent releases so I can chip away at the features one at a time. When the time comes a little closer we can vote / discuss what features people want to see.

**Reply by pg7p** [reply]
I'm really looking forward to this project. And I'm not the only one thinking.  We will react faster in the telegram group.  I think that for everyone the main goal is to get CFM and every time under equal conditions, Temperature, humidity, pressure, etc. so that the results at different times correspond

**Comment by pg7p** [comment]
I have already bought 10 vacuum cleaner motors 1800w, assembled a flow bench box, and I am waiting for a solution from you!

**Comment by DeeEmm** [comment]
Unfortunately I've been recovering from Covid which has been a long process and has seen the project pushed aside a little.. 

Prior to this I was rewriting the I2C code as what I found when testing the prototype shield was that there were issues when communicating with multiple devices over I2C. This led to data reliability issues which impacted the accuracy. 

The prototype shield is working apart from this issue but until I resolve the problem I cannot finalise the shield design as the solution may yet lay in a different hardware configuration.

I think that the issue lays with bus clashing as a result of seperate I2C comms code for each device, so I have been developing an I2C abstraction layer that solves the issue. I'm hopeful that I can get the issue solved, but I just need to finish off the new code. 

Sorry for the delays.

/DM

**Comment by dobrowolskibartosz** [comment]
Hello,
Any new updates? really looking forward to creating a bench based on this controller :)

**Comment by DeeEmm** [comment]
Sorry, I've been distracted a bit. In the middle of building works at home. 

Will try to get the I2C issue resolved this month and push out the new release.
